### DISCUSSION TITLE
Styled Components in next.js 13 out of sync

### DISCUSSION THREAD

**Discussion by codelover2k** [vercel/next.js#47926] [Help]
### Summary

I started to use next.js 13 and I am loving it 😊

I installed styled-components and followed this documentation:
[https://beta.nextjs.org/docs/styling/css-in-js#styled-components](https://beta.nextjs.org/docs/styling/css-in-js#styled-components)

My styled components work so far but what I noticed is that if I refresh my page or if I use my page for the first time, I have to wait for the styles to kick in. In my next.js 12 version I do not have this problem.

This is my **next.config.js**:
```js
/** @type {import('next').NextConfig} */

require('dotenv').config({
  ...(process.env.NODE_ENV === 'production'
    ? { path: './.env' }
    : { path: `./.${process.env.NODE_ENV}.env` }),
});

const { i18n } = require('./next-i18next.config');

const nextConfig = {
  reactStrictMode: true,
  experimental: { appDir: true },
  eslint: {
    // Warning: This allows production builds to successfully complete even if
    // your project has ESLint errors.
    ignoreDuringBuilds: true,
  },
  compiler: {
    // Enables the styled-components SWC transform
    styledComponents: true,
  },
  i18n,
};

module.exports = nextConfig;
```

this is my **layout.tsx** inside **app/**
```tsx
import StyledComponentsRegistry from './lib/registry';
import GlobalStyles from '../styles/global_style';
import Layout from '../layouts/App-Layout';


export default function RootLayout({ children }: { children: React.ReactNode }) {
  return (
    <html lang="en">
      <body>
        <StyledComponentsRegistry>
          <Layout sessionUser={{}} locale={{}}>
            <>
              <GlobalStyles />
              {children}
            </>
          </Layout>
        </StyledComponentsRegistry>
      </body>
    </html>
  );
}

```
I hope that someone can help me

**Comment by ReiDoEcommerce** [comment]
I'm having the same problem, any solution so that the css is already pre-processed by the server with the initial state and later executed in the client to be rehydrated again according to the rules?
